data-prepper-plugins/date-processor/src/main/java/com/amazon/dataprepper/plugins/processor/date/DateProcessor.java

```diff
@@ -12,37 +12,164 @@
+    private Locale sourceLocale;
+    private ZoneId sourceTimezone;
+    private List<DateTimeFormatter> dateTimeFormatters;
+    private IllegalArgumentException parsingException;
     private final DateProcessorConfig dateProcessorConfig;
 
     @DataPrepperPluginConstructor
-    protected DateProcessor(PluginMetrics pluginMetrics, final DateProcessorConfig dateProcessorConfig) throws NotImplementedException {
+    protected DateProcessor(PluginMetrics pluginMetrics, final DateProcessorConfig dateProcessorConfig) {
         super(pluginMetrics);
         this.dateProcessorConfig = dateProcessorConfig;
+
+        if (Boolean.TRUE.equals(dateProcessorConfig.getFromTimeReceived()) && dateProcessorConfig.getMatch() != null) {
+            throw new IllegalArgumentException("from_time_received and match are mutually exclusive options.");
+        }
+
+        else if (Boolean.FALSE.equals(dateProcessorConfig.getFromTimeReceived()) && dateProcessorConfig.getMatch() != null) {
+            sourceTimezone = buildZoneId(dateProcessorConfig.getTimezone());
+            if (dateProcessorConfig.getLocale() == null || dateProcessorConfig.getLocale().equalsIgnoreCase("ROOT")) {
+                sourceLocale = Locale.ROOT;
+            }
+            else {
+                sourceLocale = parseLocale(dateProcessorConfig.getLocale());
+            }
+
+            if (dateProcessorConfig.getMatch().size() != 1)
+                throw new IllegalArgumentException("match can have a minimum and maximum of 1 entry.");
+            extractKeyAndFormatters(dateProcessorConfig.getMatch());
+        }
     }
 
     @Override
     public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
+        for(final Record<Event> record : records) {
+            String sourceTimestamp;
+            String zonedDateTime = null;
+
+            if (Boolean.TRUE.equals(dateProcessorConfig.getFromTimeReceived())) {
+                Instant timeReceived = record.getData().getMetadata().getTimeReceived();
+                zonedDateTime =  getZonedDateTimeFromInstant(timeReceived);
+            }
+            else if (keyToParse != null && !keyToParse.isEmpty()) {
+                sourceTimestamp = record.getData().get(keyToParse, String.class);
+
+                for (DateTimeFormatter formatter : dateTimeFormatters) {
+                    zonedDateTime = getZonedDateTime(sourceTimestamp, formatter);
+                    if (zonedDateTime != null)
+                        break;
+                }
+                if (zonedDateTime == null) {
+                    LOG.debug("Unable to parse {} with provided patterns. {}", sourceTimestamp, parsingException.getMessage());
+                }
+            }
+            if (zonedDateTime != null)
+                record.getData().put(dateProcessorConfig.getDestination(), zonedDateTime);
+        }
+        return records;
+    }
+
+    private ZoneId buildZoneId(final String timezone) {
```

Comment:
It just seems unnecessary to have all these try catches for validation scattered around when we can focus it all in one place in the config.